src/squlearn/encoding_circuit/qcnn_encoding_circuit.py

```diff
@@ -0,0 +1,686 @@
+                if diff_params:
+                    number_of_gates_1 = int(len(self.left_qubits) / quantum_circuit.num_qubits)
+                    number_of_gates_2 = 0
+                    if alternating:
+                        number_of_gates_2 = int(
+                            (len(self.left_qubits) - 1) / quantum_circuit.num_qubits
+                        )
+                    self._num_parameters += quantum_circuit.num_parameters * (
+                        number_of_gates_1 + number_of_gates_2
+                    )
+                else:
+                    self._num_parameters += quantum_circuit.num_parameters
+                if _new_operation:
+                    self._operations_list.append(
+                        ["Conv", quantum_circuit, label, alternating, diff_params]
+                    )
+
+    def pooling(
+        self,
+        quantum_circuit: Union[QuantumCircuit, EncodingCircuitBase, None] = None,
+        label: str = "Pool",
+        measurement: bool = False,
+        input_list: list = [],
+        output_list: list = [],
+        _new_operation: bool = True,
+    ):
+        """
+        Add a pooling layer to the encoding circuit.
+
+        This reduces the number of qubits to operate on from here on in this circuit
+        by at least one for each circuit applied.
+
+        Args:
+        quantum_circuit Union[EncodingCircuitBase,QuantumCircuit, None]:
+            The quantum circuit, which is applied in this layer.
+            Must be an entangling layer, which entangles qubits.
+        label (str): The name of the layer.
+        measurement (bool): Sets whether the qubits,
+            which are not used anymore after this layer, are measured.
+            If True, quantum_circuit must consist of exactly one classical bit
+            additionally to the quantum bits.
+        input_list (list): Optionally one can pass the structure of the gates operating.
+            The input list defines the qubits the input circuit acts on.
+            The list should be structured as: [[qubit1,qubit2,..],[qubit3,qubit4,..],..].
+            Every qubit can only be adressed once and the number of qubits in each list
+            within the list must be equal to the number of qubits of input circuit.
+        output_list (list): Exactly if an input list is entered, an output list must be entered.
+            The output list defines the qubits which are left in the circuit to operate on.
+            The list should be structured as: [[qubit1,qubit2,..],[qubit3,qubit4,..],..].
+            It must have the same length as the input list and in each sublist
+            its elements must be in the corresponding input sublist
+            while beeing at least one element less.
+        THE QUBIT NUMBERS IN THE LISTS REFER TO THE INITIAL QUBIT NUMBERS!
+        Default circuit: Entangles qubit i and qubit i+1.
+            Only qubit i stays in the circuit for further operations.
```

Comment:
What is this? It seems like this should be in the description not in the argument list.